### ISSUE TITLE
What to do when hazMatrix includes negative steps?

### ISSUE THREAD

**Issue by mclements** [palryalen/transform.hazards#1]
From an `timereg::aalen` object, I am using the `predict` function to calculate the `hazMatrix` using:

    -t(log(predict(object, newdata=newdata, se=FALSE)$S0))

However, I sometimes get negative increments in `hazMatrix` and negative values for the state occupation probabilities and length of stay. Is there a standard approach for this case?

Any guidance would be appreciated.

Sincerely, Mark Clements.

**Comment by palryalen** [comment]
Hi Mark,

Thank you for using the package!

Do you adjust for covariates in the regression? In that case it could be an instance of what is sometimes called the range problem of the additive hazard estimator, i.e. negative increments of the estimated cumulative hazard trajectories. If you adjust for covariates and the sample size is large, this is an indication that your model could be better specified. Otherwise it could be a feature of having a small sample size. If you have a code sample you can send I will be happy to take a deeper look into the issue.

Best regards, Pål

**Comment by mclements** [comment]
Pål,

Thank you for this prompt reply.

The following code is an application of Aalen's additive hazards model to a multi-state analysed by Crowther and Lambert. The example shows that the predicted steps in H(t|x) are not always positive. For the estimated state occupation and length of stay calculations, this is only an issue for the first few times. I could use an identity link for the confidence intervals or simply truncate the predictions less than zero. What would you recommend?

Sincerely, Mark.

```
## Example using Crowther and Lambert (2018)
library(readstata13)
library(timereg)
mex.1 <- read.dta13("http://fmwww.bc.edu/repec/bocode/m/multistate_example.dta")
transmat <- rbind("Post-surgery"=c(NA,1,2), 
                  "Relapsed"=c(NA,NA,3),
                  "Died"=c(NA,NA,NA))
colnames(transmat) <- rownames(transmat)
mex.2 <- transform(mex.1,osi=(osi=="deceased")+0)
levels(mex.2$size)[2] <- ">20-50 mm" # fix typo
mex <- mstate::msprep(time=c(NA,"rf","os"),status=c(NA,"rfi","osi"),
                      data=mex.2,trans=transmat,id="pid",
                      keep=c("age","size","nodes","pr_1","hormon"))
mex <- transform(mex,
                 size2=(unclass(size)==2)+0, # avoids issues with TRUE/FALSE
                 size3=(unclass(size)==3)+0,
                 hormon=(hormon=="yes")+0,
                 Tstart=Tstart/12,
                 Tstop=Tstop/12)
##
c.ar <- aalen(Surv(Tstart,Tstop,status) ~ age + size2 + size3 + nodes + pr_1 + hormon,
              data = subset(mex, trans==1))
## c.ad <- aalen(Surv(Tstart, Tstop, status) ~ age + size + nodes + pr_1 + hormon,
##               data = subset(mex, trans==2))
## c.rd <- aalen( Surv(Tstart,Tstop,status) ~ age + size + nodes + pr_1 + hormon,
##               data=subset(mex, trans==3))
##
nd <- expand.grid(nodes=seq(0,20,10), size=levels(mex$size))
nd <-
    transform(nd, age=54, pr_1=3, hormon=0, size2=(unclass(size)==2)+0, size3=(unclass(size)==3)+0)
S <- predict(c.ar,newdata=nd, se=FALSE)$S0
par(mfrow=c(2,2))
matplot(t(S),type="l")
matplot(-log(t(S)),type="l")
matplot(diff(-log(t(S[,1:1450]))),type="l")
head(diff(-log(t(S))))
```

**Comment by palryalen** [comment]
Yes, the predicted H(t|x) has negative increments in the Crowther and Lambert application you provided. The sample size is quite large in this example, so I think negative increments, especially for small t, is something that should be worried about. Do you have a decent sample size in the example you are working on? I don't disagree with the remedies you suggested, but I feel you should (re)evaluate the model fit before doing anything else. Have you e.g. looked at the martingale residuals? Have you tried more than one additive hazard model, or do you have other candidate models? After all, a decreasing cumulative hazard curve at time t indicates "negative risk" at t, which doesn't make a lot of sense..
